src/Storages/System/StorageSystemFunctions.cpp

```diff
@@ -121,29 +129,40 @@ void StorageSystemFunctions::fillData(MutableColumns & res_columns, ContextPtr c
     const auto & function_names = functions_factory.getAllRegisteredNames();
     for (const auto & function_name : function_names)
     {
-        fillRow(res_columns, function_name, UInt64(0), "", FunctionOrigin::SYSTEM, functions_factory);
+        std::optional<UInt64> is_deterministic;
+        try
+        {
+            is_deterministic = functions_factory.tryGet(function_name, context)->isDeterministic();
+        }
+        catch (...)
```

Comment:
We should catch a specific exception, not everything.

Reply:
Done